### PR TITLE
Change ProjectReference adapter test package

### PR DESCRIPTION
**Pull request by NautiluX**
* Make API public
* Add tests for previously private functions

### CODE REVIEW

File: pkg/controller/projectreference/projectreference_adapter.go [hunk 1]
```diff
@@ -85,61 +85,61 @@ var supportedRegions = map[string]bool{
...
-	projectReference *gcpv1alpha1.ProjectReference
+	ProjectClaim     *gcpv1alpha1.ProjectClaim
+	ProjectReference *gcpv1alpha1.ProjectReference
 	logger           logr.Logger
 	kubeClient       client.Client
 	gcpClient        gcpclient.Client
 }
 
-func newReferenceAdapter(projectReference *gcpv1alpha1.ProjectReference, logger logr.Logger, client client.Client, gcpClient gcpclient.Client) (*ReferenceAdapter, error) {
+func NewReferenceAdapter(projectReference *gcpv1alpha1.ProjectReference, logger logr.Logger, client client.Client, gcpClient gcpclient.Client) (*ReferenceAdapter, error) {
```
Comment (drpaneas) [inline pkg/controller/projectreference/projectreference_adapter.go:95]: Needs a comment to above the function `// NewReferenceAdapater ....`

File: pkg/controller/projectreference/projectreference_adapter.go [hunk 2]
```diff
@@ -206,37 +206,37 @@ func getMatchingClaimLink(projectReference *gcpv1alpha1.ProjectReference, client
 }
 
 // updateProjectID updates the ProjectReference with a unique ID for the ProjectID
-func (r *ReferenceAdapter) updateProjectID() error {
+func (r *ReferenceAdapter) UpdateProjectID() error {
```
Comment (drpaneas) [inline pkg/controller/projectreference/projectreference_adapter.go:209]: Fix the comment `// UpdateProjectID`

speaking of it, it's not much trouble fix the comment also for https://github.com/openshift/gcp-project-operator/blob/master/pkg/controller/projectreference/projectreference_adapter.go#L276 which is referring to the wrong function.

File: pkg/controller/projectreference/projectreference_adapter.go (hunk 1 above)
Comment (NautiluX) [reply pkg/controller/projectreference/projectreference_adapter.go:95]: I dislike this so much.

File: pkg/controller/projectreference/projectreference_adapter.go (hunk 2 above)
Comment (NautiluX) [reply pkg/controller/projectreference/projectreference_adapter.go:209]: we don't need a comment for a private function, do we?

**Pull request merged**